ambry-server/src/main/java/com/github/ambry/server/AmbryServer.java

```diff
@@ -228,7 +228,7 @@ public void startup() throws InstantiationException {
 
       SSLFactory sslFactory = new NettySslHttp2Factory(sslConfig);
 
-      if (replicationConfig.replicationEnableHttp2) {
+      if (clusterMapConfig.clusterMapEnableHttp2Replication) {
```

Comment:
Do you think we will eventually use DataNodeId.getPortToConnectTo in the router too? If so, we may want to change the config name to `clusterMapEnableHttp2Client` (and deprecate the router config at some point) so that it makes sense for both the router and replication.

Reply:
Yes. 

I had an old comment:
```
  /**
   * A temporary method to get port before http2 based replication is implemented.
   * TODO: remove this once http2 based replication is implemented.
   * @param replicaId The {@link ReplicaId} to connect to.
   * @param routerEnableHttp2NetworkClient  if http2 network client should be enabled.
   * @return the port to connect.
   */
  static Port getPortToConnectTo(ReplicaId replicaId, boolean routerEnableHttp2NetworkClient) {
    if (routerEnableHttp2NetworkClient) {
      return new Port(replicaId.getDataNodeId().getHttp2Port(), PortType.HTTP2);
    } else {
      return replicaId.getDataNodeId().getPortToConnectTo();
    }
  }
```

The plan is:
1. Apply this PR and verify http2 replication works
2. Deprecate `sslEnabledDataCenters` and `routerEnableHttp2NetworkClient`. Rename `clusterMapEnableHttp2Replication` to `clusterMapEnableHttp2`. 

After deprecating `sslEnabledDataCenters`, the default connection is SSL based, if clusterMapEnableHttp2 is enabled, it's Http2 based. 

Both frontend and server will be accessible to  `clusterMapEnableHttp2`.